broadinstitute/ddp-study-server
# Improving DSM participant export performance (DDP-7581)

This massively improves the performance of export by bypassing our existing DTO/serializers.  This comes at the cost of having to reimplement much of the serialization logic in an ad-hoc way.  

This also collates all responses to REGISTRATION_X_STATE_PROVINCE questions into a single column, whereas before there was a separate column for each country

TO TEST:
(Collation)
1. Go to PanCan study participant list
2. Select all columns
3. select download, and then downoad with default options
4. Confirm that only a single "REGISTRATION_STATE_PRIVONICE" column appears for a given registration completion, and that it contains the appropriate value for the participant
5. do a search for "mailing" and confirm that the mailing address column appears correctly

(Kit requests)
1. Go to TestBoston study participant list
2. select all columns
3. select download, and then downoad with default options
4. confirm that kit requests, and test results appear correctly in the download
